### ISSUE TITLE
"SchedulerPredicates validates MaxPods" e2e test is flaky

### ISSUE THREAD

**Issue by a-robinson** [kubernetes/kubernetes#15139]
It's failed 5 times in the last 30 runs, running as part of the kubernetes-e2e-gce-slow test suite. It spits out thousands of lines of junk, with this output at the very end:
```
Expected
    <int>: 2
to equal
    <int>: 1
```

**Comment by gmarek** [comment]
I believe that #15144 should fix this, but I'll monitor the status.

My understanding is that we were counting running/non running Pods (as it was reflected in PodPhase), instead of number of scheduled/non-scheduled Pods. The problem with this approach was that sometimes some containers are crashing, falsifying the result.

**Comment by gmarek** [comment]
It's flakying again. I'll change logging to debug further.

**Comment by gmarek** [comment]
I located the problem, but I'm not sure what's happening yet:
```
19:16:04 STEP: Starting additional 291 Pods to fully saturate the cluster max pods and trying to start another one
```

Which shouldn't be the case, as we have 3 nodes, 100 max-pods limit and 10 kube-system pods:
```
elasticsearch x2
fluentd x3
heapster
kibana
kube-dns
kube-ui
influxdb
```

**Comment by alex-mohr** [comment]
5 failures in 5 days, causing 2.5 hours of submit queue blockage each time., so bumping to P0.
http://kubekins.dls.corp.google.com/view/Critical%20Builds/job/kubernetes-e2e-gce-slow/753/
http://kubekins.dls.corp.google.com/view/Critical%20Builds/job/kubernetes-e2e-gce-slow/735/
http://kubekins.dls.corp.google.com/view/Critical%20Builds/job/kubernetes-e2e-gce-slow/716/
http://kubekins.dls.corp.google.com/view/Critical%20Builds/job/kubernetes-e2e-gce-slow/714/
http://kubekins.dls.corp.google.com/view/Critical%20Builds/job/kubernetes-e2e-gce-slow/705/

/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/scheduler_predicates.go:266
Not scheduled Pods: []api.Pod{api.Pod{TypeMeta:unversioned.TypeMeta{Kind:"", APIVersion:""}, ObjectMeta:api.ObjectMeta{Name:"additional-pod", GenerateName:"", Namespace:"e2e-tests-sched-pred-cs44q", SelfLink:"/api/v1/namespaces/e2e-tests-sched-pred-cs44q/pods/additional-pod", UID:"695486d2-7dad-11e5-9935-42010af00002", ResourceVersion:"9986", Generation:0, CreationTimestamp:unversioned.Time{Time:time.Time{sec:63581658745, nsec:0, loc:(*time.Location)(0x246aee0)}}, DeletionTimestamp:(*unversioned.Time)(nil), DeletionGracePeriodSeconds:(*int64)(nil), Labels:map[string]string{"name":"additional"}, Annotations:map[string]string(nil)}, Spec:api.PodSpec{Volumes:[]api.Volume{api.Volume{Name:"default-token-1ine4", VolumeSource:api.VolumeSource{HostPath:(*api.HostPathVolumeSource)(nil), EmptyDir:(*api.EmptyDirVolumeSource)(nil), GCEPersistentDisk:(*api.GCEPersistentDiskVolumeSource)(nil), AWSElasticBlockStore:(*api.AWSElasticBlockStoreVolumeSource)(nil), GitRepo:(*api.GitRepoVolumeSource)(nil), Secret:(*api.SecretVolumeSource)(0xc208dd3fa0), NFS:(*api.NFSVolumeSource)(nil), ISCSI:(*api.ISCSIVolumeSource)(nil), Glusterfs:(*api.GlusterfsVolumeSource)(nil), PersistentVolumeClaim:(*api.PersistentVolumeClaimVolumeSource)(nil), RBD:(*api.RBDVolumeSource)(nil), Cinder:(*api.CinderVolumeSource)(nil), CephFS:(*api.CephFSVolumeSource)(nil), Flocker:(*api.FlockerVolumeSource)(nil), DownwardAPI:(*api.DownwardAPIVolumeSource)(nil), FC:(*api.FCVolumeSource)(nil)}}}, Containers:[]api.Container{api.Container{Name:"additional-pod", Image:"beta.gcr.io/google_containers/pause:2.0", Command:[]string(nil), Args:[]string(nil), WorkingDir:"", Ports:[]api.ContainerPort(nil), Env:[]api.EnvVar(nil), Resources:api.ResourceRequirements{Limits:api.ResourceList(nil), Requests:api.ResourceList(nil)}, VolumeMounts:[]api.VolumeMount{api.VolumeMount{Name:"default-token-1ine4", ReadOnly:true, MountPath:"/var/run/secrets/kubernetes.io/serviceaccount"}}, LivenessProbe:(*api.Probe)(nil), ReadinessProbe:(*api.Probe)(nil), Lifecycle:(*api.Lifecycle)(nil), TerminationMessagePath:"/dev/termination-log", ImagePullPolicy:"IfNotPresent", SecurityContext:(*api.SecurityContext)(nil), Stdin:false, StdinOnce:false, TTY:false}}, RestartPolicy:"Always", TerminationGracePeriodSeconds:(*int64)(0xc208dd3fd0), ActiveDeadlineSeconds:(*int64)(nil), DNSPolicy:"ClusterFirst", NodeSelector:map[string]string(nil), ServiceAccountName:"default", NodeName:"", SecurityContext:(*api.PodSecurityContext)(0xc208c10600), ImagePullSecrets:[]api.LocalObjectReference(nil)}, Status:api.PodStatus{Phase:"Pending", Conditions:[]api.PodCondition(nil), Message:"", Reason:"", HostIP:"", PodIP:"", StartTime:(*unversioned.Time)(nil), ContainerStatuses:[]api.ContainerStatus(nil)}}, api.Pod{TypeMeta:unversioned.TypeMeta{Kind:"", APIVersion:""}, ObjectMeta:api.ObjectMeta{Name:"maxp-290", GenerateName:"", Namespace:"e2e-tests-sched-pred-cs44q", SelfLink:"/api/v1/namespaces/e2e-tests-sched-pred-cs44q/pods/maxp-290", UID:"6629c014-7dad-11e5-9935-42010af00002", ResourceVersion:"9974", Generation:0, CreationTimestamp:unversioned.Time{Time:time.Time{sec:63581658740, nsec:0, loc:(*time.Location)(0x246aee0)}}, DeletionTimestamp:(*unversioned.Time)(nil), DeletionGracePeriodSeconds:(*int64)(nil), Labels:map[string]string{"name":"maxp-290"}, Annotations:map[string]string(nil)}, Spec:api.PodSpec{Volumes:[]api.Volume{api.Volume{Name:"default-token-1ine4", VolumeSource:api.VolumeSource{HostPath:(*api.HostPathVolumeSource)(nil), EmptyDir:(*api.EmptyDirVolumeSource)(nil), GCEPersistentDisk:(*api.GCEPersistentDiskVolumeSource)(nil), AWSElasticBlockStore:(*api.AWSElasticBlockStoreVolumeSource)(nil), GitRepo:(*api.GitRepoVolumeSource)(nil), Secret:(*api.SecretVolumeSource)(0xc208cefeb0), NFS:(*api.NFSVolumeSource)(nil), ISCSI:(*api.ISCSIVolumeSource)(nil), Glusterfs:(*api.GlusterfsVolumeSource)(nil), PersistentVolumeClaim:(*api.PersistentVolumeClaimVolumeSource)(nil), RBD:(*api.RBDVolumeSource)(nil), Cinder:(*api.CinderVolumeSource)(nil), CephFS:(*api.CephFSVolumeSource)(nil), Flocker:(*api.FlockerVolumeSource)(nil), DownwardAPI:(*api.DownwardAPIVolumeSource)(nil), FC:(*api.FCVolumeSource)(nil)}}}, Containers:[]api.Container{api.Container{Name:"maxp-290", Image:"beta.gcr.io/google_containers/pause:2.0", Command:[]string(nil), Args:[]string(nil), WorkingDir:"", Ports:[]api.ContainerPort(nil), Env:[]api.EnvVar(nil), Resources:api.ResourceRequirements{Limits:api.ResourceList(nil), Requests:api.ResourceList(nil)}, VolumeMounts:[]api.VolumeMount{api.VolumeMount{Name:"default-token-1ine4", ReadOnly:true, MountPath:"/var/run/secrets/kubernetes.io/serviceaccount"}}, LivenessProbe:(*api.Probe)(nil), ReadinessProbe:(*api.Probe)(nil), Lifecycle:(*api.Lifecycle)(nil), TerminationMessagePath:"/dev/termination-log", ImagePullPolicy:"IfNotPresent", SecurityContext:(*api.SecurityContext)(nil), Stdin:false, StdinOnce:false, TTY:false}}, RestartPolicy:"Always", TerminationGracePeriodSeconds:(*int64)(0xc208cefee0), ActiveDeadlineSeconds:(*int64)(nil), DNSPolicy:"ClusterFirst", NodeSelector:map[string]string(nil), ServiceAccountName:"default", NodeName:"", SecurityContext:(*api.PodSecurityContext)(0xc2084c1780), ImagePullSecrets:[]api.LocalObjectReference(nil)}, Status:api.PodStatus{Phase:"Pending", Conditions:[]api.PodCondition(nil), Message:"", Reason:"", HostIP:"", PodIP:"", StartTime:(*unversioned.Time)(nil), ContainerStatuses:[]api.ContainerStatus(nil)}}}
Expected
    <int>: 2
to equal
    <int>: 1

**Comment by davidopp** [comment]
I and almost all of the Poland team are OOO this week; I'd suggest disabling this test for now if it is blocking submits. (If it was consistently failing I'd be concerned, but you said it's just flaky.)

**Comment by gmarek** [comment]
I tried reproducing the problem locally but failed completely - I run the test 20 times, and it succeeded every single time.

**Comment by gmarek** [comment]
I believe it's a crashloop problem. I the last failure (slow suite run 873) at the beginning we have:
```
Nov 16 21:33:54.118: INFO: 11 / 11 pods in namespace 'kube-system' are running and ready (32 seconds elapsed)
```
But during the MaxPods run we see:
```
STEP: Starting additional 290 Pods to fully saturate the cluster max pods and trying to start another one
```

Which means that during the cluster stabilization phase we didn't saw one system pod. In addition during 'Resize' test there's a line:
```
Nov 16 22:15:26.051: INFO: 10 / 10 pods in namespace 'kube-system' are running and ready (0 seconds elapsed)
```

Which suggests that we're not really returning to the original state after resize test.

**Comment by gmarek** [comment]
Day one with #17687 merged - no failures yet.

**Comment by gmarek** [comment]
Day two without failures.

**Comment by gmarek** [comment]
It's been 6 days without failures. Assuming that the issue was fixed by #17678.
